### ISSUE TITLE
sources url in the packaged html files is incorrect

### ISSUE THREAD

**Issue by JZLeung** [parcel-bundler/parcel#405]
my project has multiple pages

index.html can jump to second.html

if i set the `--public-url` as `./`, the packged files can work well

but if i set the `--public-url` as a relative path, like `public`, the second.html will not be able to introduce the correct resources, because the js file is in `example.com/public/xxx.js` instead of `example.com/public/public/xxx.js`
```
GET http://192.168.2.3:8000/public/public/1fad9c454e4ab4186605464b35256085.js 404 (Not Found)
``` 
I think all the html files can be placed in the project root directory, not in the resource directory.

**Comment by robinp** [comment]
Slightly related problem I stumbled upon (parcel 1.3.1).

```
$ cat index.html
<html>
<body>
 <script src="./output/Main/index.js"></script>
</body>
</html>
```
```
$ parcel --public-url 'http://rigo:8000' -p 8000 -d dist index.html
Server running at http://localhost:8000
Built in 4.70s.
```

```
$ curl 'http://rigo:8000/'
<html>
<body>
 <script src="http:/rigo:8000/4bfee3649edec6cdb33724f88316be5d.js"></script>
</body>
</html>
```

```
$ curl 'http:/rigo:8000/4bfee3649edec6cdb33724f88316be5d.js'
<html>
<body>
 <script src="http:/rigo:8000/4bfee3649edec6cdb33724f88316be5d.js"></script>
</body>
</html>
```

So the js path served in the main html is not correct (I tried other seemingly trivial paths, like `/dists/....js`, without success). If I try without `--public-url`, it works.

**Comment by devongovett** [comment]
Should be fixed by #447.
